webui/react/src/hooks/useModal/useModal.tsx

```diff
@@ -73,7 +73,7 @@ const useModal = (
             resolve(thenArgs);
             modalClose(reason);
           })
-          .catch(reject);
+          .catch(() => reject());
```

Comment:
this was required for some reason. The behavior is completely different. As `.catch(reject)` when errors are thrown, the `reject` function does NOT get called properly. 🤷

Reply:
I also noticed this behavior for `useEffect` return callbacks.

Reply:
hmm yeah [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) doesn't show any new `.catch` call signature I'm not aware of either.
should we still pass in the error reason when we call reject?